packages/app/src/systems/Asset/components/AssetInput/AmountInput.tsx

```diff
@@ -0,0 +1,111 @@
+import { cssObj } from '@fuel-ui/css';
+import { Button, Flex, Input, Text } from '@fuel-ui/react';
+import { DECIMAL_UNITS, bn } from 'fuels';
+import type { BN } from 'fuels';
+import { useState } from 'react';
+import type { FC, ChangeEvent } from 'react';
+
+import { AmountInputLoader } from './AmountInputLoader';
+
+export type AmountInputProps = {
+  balance: BN;
+  value: BN;
```

Comment:
I don't see why `value` prop should be an `BN`, you have a specific reason for that?

if not let's change to `value: string;`

it makes more sense because:
1- BN don't support floats , decimals.
2- BN is made to work with units, which is not the case here. You're working with regular decimals in this case
3- who's using the component, if got a value from API for example, will inform: `value={amount.format()}`, then the string already starts formatted here in the exact way the user inputed.

Reply:
I changed it bc of @luizstacio requested it in his above comment.  I may have misinterpreted his comment though.

Reply:
ok let's w8 for his answer

Reply:
if you we don't abstract but pass just a string all the logic to handle the BN needs to be done on each screen. This is why I think receiving the BN is smarter on this case.

Reply:
@luizstacio it really makes sense to keep as value as `BN`.

`onChange` also sends a `BN` which should be consistent with `value` input